controllers/data.controller.js

```diff
@@ -118,33 +132,70 @@ const handleClassicQuestions = async (
   if (updatedScore < -50) {
     newScore = 100 + updatedScore; // score is negative so we add it to 100 (ex : 100 + (-50) = 50)
     newTier = TIER_LIST[tierIndex - 1];
-  } else if (updatedScore >= 100) {
+  }
+  // update score and tier if user has enough points and is not master or above
+  else if (updatedScore >= 100 && tierIndex < TIER_LIST.length - 3) {
     newScore = updatedScore - 100;
     newTier = TIER_LIST[tierIndex + 1];
+    if (newTier === "M") {
+      newScore = 0;
+    }
   }
-  if (uid === undefined) {
-    response = {
-      tier: newTier,
-      score: newScore,
-      questionsAnsweredNb: questionsAnsweredNb + 1,
-    };
-  } else {
-    response = await User.findOneAndUpdate(
-      { _id: uid },
-      {
-        $set: {
-          score: newScore,
-          tier: newTier,
-          questionsAnsweredNb: questionsAnsweredNb + 1,
-        },
+
+  response = await User.findOneAndUpdate(
+    { _id: uid },
+    {
+      $set: {
+        score: newScore,
+        tier: newTier,
+        questionsAnsweredNb: questionsAnsweredNb + 1,
       },
-      { returnOriginal: false },
-    );
+    },
+  );
+  if (["M", "GM", "C"].includes(newTier)) {
+    response = await updateTopTiers(response);
   }
-
   return response;
 };
 
+const updateTopTiers = async (currentUser) => {
+  const topUsers = await User.find({ tier: { $in: ["M", "GM", "C"] } }).sort({
+    score: "desc",
+  });
+  let updatedUser = currentUser;
+  topUsers.forEach(async (user, index) => {
```

Comment:
Using async/await with forEach loop in JavaScript can be a bit tricky since forEach doesn’t inherently support asynchronous operations, so use for loop instead